pkg/controller/controller_utils.go

```diff
@@ -430,14 +462,17 @@ func GetPodFromTemplate(template *api.PodTemplateSpec, parentObject runtime.Obje
 			GenerateName: prefix,
 		},
 	}
+	if controllerRef != nil {
+		pod.OwnerReferences = append(pod.OwnerReferences, *controllerRef)
```

Comment:
I'm not sure I follow you. CreatePodsWithControllerRef doesn't create the pod, it passes the controllerRef to here to create a pod with controllerRef.